auth/integration_test/src/integration_test.cc

```diff
@@ -719,62 +719,75 @@ TEST_F(FirebaseAuthTest, TestWithCustomEmailAndPassword) {
-  // Save the old provider ID list so we can make sure it's the same once it's
-  // loaded again.
-  std::vector<std::string> prev_provider_data_ids;
-  for (int i = 0; i < auth_->current_user()->provider_data().size(); i++) {
-    prev_provider_data_ids.push_back(
-        auth_->current_user()->provider_data()[i]->provider_id());
-  }
-  Terminate();
-  ProcessEvents(2000);
-  Initialize();
-  EXPECT_NE(auth_, nullptr);
-  ASSERT_NE(auth_->current_user(), nullptr);
-  EXPECT_FALSE(auth_->current_user()->is_anonymous());
-  // Make sure the provider IDs are the same as they were before.
-  EXPECT_EQ(auth_->current_user()->provider_id(), prev_provider_id);
-  std::vector<std::string> loaded_provider_data_ids;
-  for (int i = 0; i < auth_->current_user()->provider_data().size(); i++) {
-    loaded_provider_data_ids.push_back(
-        auth_->current_user()->provider_data()[i]->provider_id());
+  // Test is disabled on linux due to the need to unlock the keystore.
+  SKIP_TEST_ON_LINUX;
+
+  if (!RunFlakyBlock([&]() {
+        std::string email = GenerateEmailAddress();
+        FLAKY_WAIT_FOR_COMPLETION(
+            auth_->CreateUserWithEmailAndPassword(email.c_str(), kTestPassword),
+            "CreateUserWithEmailAndPassword");
+        FLAKY_EXPECT_NONNULL(auth_->current_user());
+        FLAKY_EXPECT_FALSE(auth_->current_user()->is_anonymous());
+        std::string prev_provider_id = auth_->current_user()->provider_id();
+        // Save the old provider ID list so we can make sure it's the same once
+        // it's loaded again.
+        std::vector<std::string> prev_provider_data_ids;
+        for (int i = 0; i < auth_->current_user()->provider_data().size();
+             i++) {
+          prev_provider_data_ids.push_back(
+              auth_->current_user()->provider_data()[i]->provider_id());
+        }
+        Terminate();
+        ProcessEvents(2000);
+        Initialize();
+        FLAKY_EXPECT_NONNULL(auth_);
+        FLAKY_EXPECT_NONNULL(auth_->current_user());
+        FLAKY_EXPECT_FALSE(auth_->current_user()->is_anonymous());
+        // Make sure the provider IDs are the same as they were before.
+        FLAKY_EXPECT_EQ(auth_->current_user()->provider_id(), prev_provider_id);
+        std::vector<std::string> loaded_provider_data_ids;
+        for (int i = 0; i < auth_->current_user()->provider_data().size();
+             i++) {
+          loaded_provider_data_ids.push_back(
+              auth_->current_user()->provider_data()[i]->provider_id());
+        }
+        FLAKY_EXPECT_TRUE(loaded_provider_data_ids == prev_provider_data_ids);
+
+        // Cleanup, ensure we are signed in as the user so we can delete it.
+        FLAKY_WAIT_FOR_COMPLETION(
+            auth_->SignInWithEmailAndPassword(email.c_str(), kTestPassword),
+            "SignInWithEmailAndPassword");
+        FLAKY_EXPECT_NONNULL(auth_->current_user());
+        DeleteUser();
+
+        return true;
```

Comment:
Consider replacing this with something like:

    return FLAKEY_SUCCESS;

or something similar.

Reply:
I just made it FLAKY_SUCCESS() to match the syntax of the others.